src/services/getExpectedApy.ts

```diff
@@ -11,16 +11,48 @@ export const getExpectedApy = (
   vt: number, // variable token balance
   margin: number, // margin until the current timestamp (margin + accrued cashflow)
   predictedApr: number, // the predicted variable APY between current and end
+  rateOracleID: number, // the rate oracle ID that suggests what type of APY is used (linear vs. compounding)
 ): [number, number] => {
-  // (rate[end]/rate[current])^(YEAR / (end-current)) - 1 = APY(current, end)
-  // rate[end]/rate[current] = (APY(current, end) + 1)^((end - current)/YEAR)
-  // rate[end]/rate[current] - 1 = (APY(current, end) + 1)^((end - current)/YEAR) - 1
-  const vf = (1 + predictedApr) ** getAnnualizedTime(current, end) - 1;
+  if (end <= current) {
+    return [0, 0];
+  }
+
+  let vf = 0;
+
+  switch (rateOracleID) {
+    case 1:
+    case 2:
+    case 3:
+    case 4:
+    case 5:
+    case 6:
+    case 7: {
+      // (rate[end]/rate[current])^(YEAR / (end-current)) - 1 = APY(current, end)
+      // rate[end]/rate[current] = (APY(current, end) + 1)^((end - current)/YEAR)
+      // rate[end]/rate[current] - 1 = (APY(current, end) + 1)^((end - current)/YEAR) - 1
+      vf = (1 + predictedApr) ** getAnnualizedTime(current, end) - 1;
+
+      break;
+    }
+
+    case 8: {
+      // (rate[end]/rate[current]-1) * (YEAR / (end - current)) = APY(current, end)
+      // rate[end]/rate[current] = APY(current, end) * (end - current) / YEAR + 1
+      // rate[end]/rate[current] - 1 = APY(current, end) * (end - current) / YEAR
+      vf = predictedApr * getAnnualizedTime(current, end);
+
+      break;
+    }
+
+    default: {
+      throw new Error('Unrecognized protocol');
+    }
+  }
 
   // cashflow = uft * (end - current) / YEAR * 0.01 + vt * (rate[end]/rate[current] - 1)
   const ecs = uft * getAnnualizedTime(current, end) * 0.01 + vt * vf;
 
-  // PNL = (1 + estimated cashflow / margin so far) ^ (YEAR / (end - current)) - 1
+  // PNL = (estimated cashflow / margin so far) * (YEAR / (end - current))
   const pnl = (ecs / margin) * (1 / getAnnualizedTime(current, end));
```

Comment:
indeed, looks like it uses simple interest formula in here, btw think worth renaming pnl to expected_apy since that's what it is calculating i think? pnl in my head at least is a cashflow measure

Reply:
sure